src/components/uielements/assets/assetCard/AssetCardMenu.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useMemo, useCallback } from 'react'
+
+import { validBNOrZero } from '@thorchain/asgardex-util'
+import { get as _get } from 'lodash'
+
+import { getTickerFormat } from '../../../../helpers/stringHelper'
+import { PriceDataIndex } from '../../../../services/midgard/types'
+import { AssetPair } from '../../../../types/asgardex.d'
+import CoinData from '../../coins/coinData'
+import FilterMenu from '../../filterMenu'
+
+const filterFunction = (item: AssetPair, searchTerm: string) => {
+  const tokenName = getTickerFormat(item.asset)
+  return tokenName?.toLowerCase().indexOf(searchTerm.toLowerCase()) === 0
+}
+
+type Props = {
+  asset: string
+  assetData: AssetPair[]
+  priceIndex?: PriceDataIndex
+  unit: string
+  searchDisable: string[]
+  withSearch: boolean
+  onSelect: (value: string) => void
+  'data-test'?: string
+}
+
+const AssetCardMenu: React.FC<Props> = (props: Props): JSX.Element => {
+  const {
+    assetData,
+    asset,
+    priceIndex = {},
+    unit,
+    withSearch,
+    searchDisable = [],
+    onSelect = () => {},
+    'data-test': dataTest = '',
+    ...otherProps // (Rudi) need to pass props for antd to provide box shadow
```

Comment:
Pls don't use spread syntax for any properties here and anywhere else - that's just unsafe. I know we did this in BEPSwap...